core/src/main/java/org/neo4j/graphalgo/api/GraphSetup.java

```diff
@@ -0,0 +1,94 @@
+package org.neo4j.graphalgo.api;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * DTO to ease the use of the GraphFactory-CTor. Should contain
+ * setup options for loading the graph from neo4j.
+ *
+ * @author mknblch
+ */
+public class GraphSetup {
+
+    // start label type. null means any label.
+    public final String startLabel;
+    // end label type (not yet implemented).
+    public final String endLabel;
+    // relationtype name. null means any relation.
+    public final String relationshipType;
+    // propertyName. null means NO property (the default value will be used instead).
+    public final String propertyName;
+    // default property is used for weighted relationships if property is not set.
+    public final double propertyDefaultValue;
+    // the executor service for parallel execution. null means single threaded evaluation.
+    public final ExecutorService executor;
```

Comment:
Not sure if mixing the Executor with declarative graph information is such a good idea.
We can leave it in for now, but should pull it out later.